pkg/remote/resolution/request.go

```diff
@@ -30,3 +30,15 @@ type resolutionRequest struct {
 func (req *resolutionRequest) OwnerRef() metav1.OwnerReference {
 	return *kmeta.NewControllerRef(req.owner)
 }
+
+var _ resolution.RequestV2 = &resolutionRequestV2{}
+var _ resolution.OwnedRequest = &resolutionRequestV2{}
+
+type resolutionRequestV2 struct {
```

Comment:
Some notes on naming - there seem to be 3 different resolution requests:

- resolution.resolutionRequest (this, pkg/remote/resolution)
- common.Request (pkg/resolution/common)
- v1beta1.ResolutionRequestSpec

They all have similar names, and it's not immediately obvious at first glance how they differ. Can we rename / consolidate these?